### PR TITLE
closes #996 - `generate_parameter_configs` adding `use_masked_lm` and `drop_last`

### LINKED ISSUE #996: `test_synthetic_lm_data.generate_parameter_configs` missing `'drop_last'`and `'use_masked_lm'` needed for `LMDatasetHparams`
[`test_synthetic_lm_data.generate_parameter_configs`](https://github.com/mosaicml/composer/blob/fa39a3150bc7d340dae4c4737901e6cffa4454c1/tests/datasets/test_synthetic_lm_data.py#L8-L24) doesn't include `LMDataSetHparams` `"drop_last"` and `"use_masked_lm"`.

The default values for `LMDatasetHparams` are:

[`drop_last = True`](https://github.com/mosaicml/composer/blob/847f31abe4bee4bc3e351e7c1a75cd3c87fc692a/composer/datasets/hparams.py#L71-L74)

[`"use_masked_lm"=None` which throws an error, since it must be set.](https://github.com/mosaicml/composer/blob/a264ae95a5a9658e1f82206cf99158281c6dd8ca/composer/datasets/lm_datasets.py#L69-L70)

## Proposal 

Set  `drop_last = False` and config['use_masked_lm'] = config['tokenizer_family'] == 'bert' as with our [gpt2](https://github.com/mosaicml/composer/blob/847f31abe4bee4bc3e351e7c1a75cd3c87fc692a/composer/yamls/models/gpt2_1%2C3b.yaml#L8-L10) and [bert](hhttps://github.com/mosaicml/composer/blob/847f31abe4bee4bc3e351e7c1a75cd3c87fc692a/composer/yamls/models/bert-base.yaml#L27-L29) yamls.
        

## To reproduce

```
from composer.models import BERTHparams
from tests.common.models import generate_dummy_model_config
from tests.datasets import test_synthetic_lm_data
from composer.datasets.synthetic_lm import generate_synthetic_tokenizer, synthetic_hf_dataset_builder

lm_dataset_configs = [config[0] for config in test_synthetic_lm_data.generate_parameter_configs( ['num_samples', 'chars_per_sample', 'column_names', 'tokenizer_family'])]
for config in lm_dataset_configs[:2]:
  print('config['tokenizer_family']')
  dataloader = LMDatasetHparams(use_synthetic=True, tokenizer_name=config['tokenizer_family'],  split='train', max_seq_length=config["chars_per_sample"])
  print(dataloader.drop_last)
  print(dataloader.use_masked_lm)
  print(dataloader.mlm_probability)
  print()
```

## Expected behavior

```
bert
False
True
0.15

gpt2
False
False
0.15
```

## Additional context

#992 manually fixes this in by [adding to the config dict](https://github.com/mosaicml/composer/blob/90d0f29a66fdabb2ba9c8ffe1c042fb645766ac7/tests/fixtures/synthetic_hf_state.py#L13-L16)

This issue has been discussed based on an earlier comment by @moinnadeem  in #992

### PR DESCRIPTION
**Pull request by jzf2101**
See #996

### CODE REVIEW

File: tests/datasets/test_synthetic_lm_data.py [hunk 1]
```diff
@@ -20,7 +20,11 @@ def generate_parameter_configs(keys, num_replicas=1):
 
     config_combinations = []
     for combo in product(*[config_options[i] for i in keys]):
-        config_combinations.append([dict(zip(keys, combo)) for _ in range(num_replicas)])
+        config = dict(zip(keys, combo))
+        if "tokenizer_family" in keys:
```
Comment (moinnadeem) [inline tests/datasets/test_synthetic_lm_data.py:24]: When would this not be true?

**Review by moinnadeem — APPROVED** [review]
LGTM!

File: tests/datasets/test_synthetic_lm_data.py (hunk 1 above)
Comment (jzf2101) [reply tests/datasets/test_synthetic_lm_data.py:24]: you mean when do we not pass `tokenizer_family` in the keys argument? It's sometimes not passed in one of the tests.  Not sure why but it's here https://github.com/mosaicml/composer/blob/fb4beb134ac9f2b88cb562e1ad06e19686092289/tests/datasets/test_synthetic_lm_data.py#L46

Comment (jzf2101) [reply tests/datasets/test_synthetic_lm_data.py:24]: resolved per @moinnadeem

**Pull request merged**